src/chi-vue/src/components/transfer-list/TransferListActions.vue

```diff
@@ -1,18 +1,135 @@
+      },
+    ],
+    sort: [
+      {
+        icon: 'chevron-up',
+        event: () => !isSortButtonDisabled('up') && onSortItems('up'),
+        disabled: () => isSortButtonDisabled('up'),
+      },
+      {
+        icon: 'chevron-down',
+        event: () => !isSortButtonDisabled('down') && onSortItems('down'),
+        disabled: () => isSortButtonDisabled('down'),
+      },
+    ],
+  };
+
+  return actions[props.move];
+};
+
+const handleTransferItems = (direction: 'from' | 'to') => {
+  const items = selectedItems.value[direction];
+  const filteredList = transferList.value.filter(({ value }) => !items.includes(value));
+  const itemsToMove = getCurrentList(direction)
+    .filter(({ value }) => items.includes(value))
+    .map((item) => ({ ...item, selected: !item.selected }));
+  const newList = [...filteredList, ...itemsToMove];
+
+  onUpdateTransferList(newList);
+  onClearSelection();
+};
+
+const handleTransferAll = (direction: 'from' | 'to') => {
+  const currentColumn = getCurrentList(direction);
+  const targetColumn = getCurrentList(direction === 'from' ? 'to' : 'from');
+  const lockedItems = currentColumn.filter(({ locked }) => locked);
+  const unlockedItems = currentColumn.filter(({ locked }) => !locked);
+  const mappedItems = unlockedItems.map((item) => ({ ...item, selected: !item.selected }));
+  const newList = [...targetColumn, ...lockedItems, ...mappedItems];
+
+  onUpdateTransferList(newList);
+  onClearSelection();
+};
+
+const onSortItems = (direction: 'up' | 'down') => {
+  const itemsToMove = selectedItems.value.to;
+  const sorted = sortTransferList({ direction, items: itemsToMove });
+  const newList = [...getCurrentList('from'), ...sorted.flat()];
+
+  onUpdateTransferList(newList);
+};
+
+const getItemStatus = (item: string, direction: 'up' | 'down') => {
+  const currItemIndex = getCurrentList('to').findIndex(({ value }) => value === item);
+  const currItem = getCurrentList('to')[currItemIndex] as TransferListItem;
+  const nextItemIndex = direction === 'up' ? currItemIndex - 1 : currItemIndex + 1;
+  const nextItem = getCurrentList('to')[nextItemIndex] as TransferListItem;
+  const isDisabled = !nextItem || (nextItem.locked && !currItem?.wildcard);
+
+  return { currItemIndex, currItem, nextItemIndex, nextItem, isDisabled };
+};
+
+const sortTransferList = ({ direction, items }): TransferListItem[] => {
+  const list = getCurrentList('to');
+
+  items.forEach((item: string) => {
+    const { currItemIndex, nextItemIndex, isDisabled } = getItemStatus(item, direction);
+
+    if (!isDisabled) {
+      swapElementsInArray<TransferListItem>(list, currItemIndex, nextItemIndex);
+    }
+  });
+
+  return list;
+};
+
+const isSortButtonDisabled = (direction: 'up' | 'down') => {
+  const selectedItem = selectedItems.value.to[0];
+  const { isDisabled } = getItemStatus(selectedItem, direction);
+
+  return selectedItem ? isDisabled : true;
+};
+
+function getCurrentList(column: 'from' | 'to' = 'from') {
```

Comment:
What is the criteria you are following to define the functions? can they be all either constants or functions, but not mixed up?